caas/kubernetes/provider/application/constraints.go

```diff
@@ -266,15 +281,34 @@ func processPodAffinity(pod *core.PodSpec, affinityLabels map[string]string) err
 	return nil
 }
 
-func configureConstraint(pod *core.PodSpec, resourceName core.ResourceName, value string) (err error) {
+func configureCharmConstraint(pod *core.PodSpec, resourceName core.ResourceName, requestValue, limitValue string) (err error) {
 	if len(pod.Containers) == 0 {
 		return nil
 	}
-	pod.Containers[0].Resources.Requests, err = MergeConstraint(resourceName, value, pod.Containers[0].Resources.Requests)
-	if err != nil {
-		return errors.Annotatef(err, "merging request constraint %s=%s", resourceName, value)
+	for i, container := range pod.Containers {
+		if container.Name == "charm" {
```

Comment:
Use the const

Reply:
Also, invert it so that if != charm continue

Reply:
will change this thanks ian!